names/README.md

```diff
@@ -13,37 +13,104 @@ Consumers of `names` include:
-| 17 | **HumanFriendly** | Code | [REQUIRED] Human-friendly name of service as used by AWS; documentation `subcategory` must exactly match this value; used in website navigation and error messages |
-| 18 | **Brand** | Code | Either `Amazon`, `AWS`, or blank (rare) as used by AWS; used in error messages |
-| 19 | **Exclude** | Code | Whether the service should be included; if included (blank), **ProviderPackageActual** or **ProviderPackageCorrect** must have a value |
-| 20 | **NotImplemented** | Code | Whether the service is implemented by the provider |
-| 21 | **EndpointOnly** | Code | If **NotImplemented** is non-blank, whether the service endpoint should be included in the provider `endpoints` configuration |
-| 22 | **AllowedSubcategory** | Code | If **Exclude** is non-blank, whether to include **HumanFriendly** in `website/allowed-subcategories.txt` anyway. In other words, if non-blank, overrides **Exclude** in some situations. Some excluded pseudo-services (_e.g._, VPC is part of EC2) are still subcategories. Only applies if **Exclude** is non-blank. |
-| 23 | **DeprecatedEnvVar** | Code | Deprecated `AWS_<service>_ENDPOINT` envvar defined for some services |
-| 24 | **TFAWSEnvVar** | Code | `TF_AWS_<service>_ENDPOINT` envvar defined for some services |
-| 25 | **Note** | Reference | Very brief note usually to explain why excluded |
+After any edits to `data/names_data.hcl`, run `make gen`. Doing so regenerates code and performs checks on `data/names_data.hcl`.
+
+The schema of the attributes and blocks of `data/names_data.csv` are as follows:
+```hcl
+service "" {
+
+  // If both of these attributes are the same as the service block's name, this block will be ommitted
+  cli_v2_command { 
+    aws_cli_v2_command           = ""
+    aws_cli_v2_command_no_dashes = ""
+  }
+
+  // If both of these attributes are the same as the service block's name, this block will be ommitted
+  go_packages { 
+    v1_package = ""
+    v2_package = ""
+  } 
+
+  sdk {
+    id             = "" 
+    client_version = [] 
+  }
+
+  names {
+    aliases             = [""]
+    provider_name_upper = ""
+    human_friendly      = ""
+  }
+
+  client {
+    go_v1_client_typename = ""
+    skip_client_generate  = bool
+  }
+
+  env_var {
+    deprecated_env_var = ""
+    tf_aws_env_var     = ""
+  }
+
+  endpoint_info {
+    endpoint_api_call        = ""
+    endpoint_api_params      = ""
+    endpoint_region_override = ""
+    endpoint_only            = bool
+  }
+
+  resource_prefix {
+    actual  = ""
+    correct = ""
+  }
+
+  provider_package_correct = ""
+  split_package       = ""
+  file_prefix         = ""
+  doc_prefix          = [""]
+  brand               = ""
+  exclude             = bool
+  not_implemented     = bool
+  allowed_subcategory = bool
+  note                = ""
+}
+
+The explanation of the attributes of `data/names_data.csv` are as follows:
+
+```
```

Comment:
I think these are swapped?

```suggestion
```

The explanation of the attributes of `data/names_data.csv` are as follows:
```